demos/js/messageMonitor.js

```diff
@@ -0,0 +1,161 @@
+                options: {
+                    model: {
+                        portSpecs: [{
+                            name: "Launchpad Pro .+ (MIDI|Standalone Port)"
+                        }]
+                    },
+                    components: {
+                        portConnector: {
+                            options: {
+                                dynamicComponents: {
+                                    connection: {
+                                        options: {
+                                            // Listen at this level so that we can determine which port/device sent the message.
+                                            listeners: {
+                                                "onMessage.log": {
+                                                    funcName: "youme.demos.messageMonitor.logMessage",
+                                                    args: ["{youme.demos.messageMonitor}", "{that}", "{arguments}.0"] // messageMonitor, connection, message
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            log: {
+                type: "youme.demos.messageMonitor.logger",
+                container: "{that}.dom.log",
+                options: {
+                    model: {
+                        loggedMessages: "{youme.demos.messageMonitor}.model.loggedMessages"
+                    }
+                }
+            },
+            resetButton: {
+                type: "fluid.viewComponent",
+                container: "{that}.dom.reset",
+                options: {
+                    listeners: {
+                        "onCreate.bind": {
+                            this: "{that}.container",
+                            method: "click",
+                            args: ["{youme.demos.messageMonitor}.resetLog"]
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    youme.demos.messageMonitor.logMessage = function (messageMonitor, connection, message) {
+        var date = new Date();
+        var timestampOptions = {
+            hour: "2-digit",
+            minute: "2-digit",
+            hour12: false,
+            fractionalSecondDigits: 2
+        };
+        var timestamp = date.toLocaleTimeString(navigator.language, timestampOptions);
+        var toLog = { timestamp: timestamp, port: connection.model.port, message: message};
+
+        // Incrementally add entries rather than deleting and recreating every time.
+        var index = messageMonitor.model.loggedMessages.length;
+        messageMonitor.applier.change(["loggedMessages", index], toLog);
+    };
+
+    youme.demos.messageMonitor.resetLog = function (messageMonitor) {
```

Comment:
There is now a standard utility
````
fluid.replaceModelValue(messageMonitor.applier, [], [])
````
which will do this in one step

Reply:
Cheers, I went through the code and replaced the pattern with the new function throughout.  I left the transactions (in the tests) which only delete the value alone.